realm/src/main/java/io/realm/RealmList.java

```diff
@@ -236,8 +236,8 @@ public int size() {
      */
     public RealmQuery<E> where() {
         if (managedMode) {
-            TableQuery query = this.view.where();
-            return new RealmQuery<E>(this.realm, query, clazz);
+            TableOrView table = this.view.where().findAll();
```

Comment:
We should definitely not do a query in where(). Youmust be able to build up a query and then execute it afterwards.